### DISCUSSION TITLE
How to update db after renaming a collection name or somthing like that?

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#1784] [Q&A] [answered]
Hi
So, I renamed a bonch of collection names and stuff like that, after saving it said you need to do some stuff manually... How can I do that? Admin UI works after renaming but it seem like the stored data as not updated...
I also have the question, why it's not updating the stuff automatically after renaming...?
Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
It depends on the type of the change that you've made.

Per the warning in the UI, when renaming a field, you have to manually update the field name in any collections rules or filters (**including in your client side code**; _the collection api rules changes eventually will be automated in the future_).

If you have changed a constraint, eg. "Nonempty", "Unique", min, max, etc. it will not update the old data because we don't know what value the old data should have and this is left to the developers to handle on their own if they need to via migrations - https://pocketbase.io/docs/migrations/. Currently migrations can be written in both Go and JS, but the JS documentations for now is postponed, since I prioritized other tasks (_for an example you can also check https://github.com/pocketbase/pocketbase/discussions/1464_).

Feel free to provide more details what you've changed and what you need to migrate and I'll try to help with the migration script.
